vhdbuilder/packer/pre-install-dependencies.sh

```diff
@@ -89,4 +89,36 @@ if [[ ${OS} == ${MARINER_OS_NAME} ]] && [[ "${ENABLE_CGROUPV2,,}" == "true" ]];
   enableCgroupV2forAzureLinux
 fi
 
+if [[ "${UBUNTU_RELEASE}" == "22.04" ]]; then
+  echo "Logging the currently running kernel: $(uname -r)"
+  echo "Before purging kernel, here is a list of kernels/headers installed:"; dpkg --list | grep 'linux-image\|linux-headers'
+
+  # Get the list of current kernel packages and headers
+  kernel_packages=($(dpkg --list | grep 'linux-image\|linux-headers' | awk '{print $2}'))
+  packages_to_remove=""
+
+  # Iterate through the kernel packages and headers and add ALL to remove list
+  for package in "${kernel_packages[@]}"; do
+      packages_to_remove+=" $package"
+  done
+
+  # Remove the unwanted kernel packages and headers
+  if [ -n "$packages_to_remove" ]; then
+      echo "Removing the following packages:$packages_to_remove"
+      apt-get --purge -y $packages_to_remove
+      apt-get autoremove
+      echo "Kernel cleanup completed."
+  else
+      echo "No packages to remove. Keeping specified kernel versions."
+  fi
+
+  echo "After purging kernel, dpkg list sould be empty"; dpkg --list | grep 'linux-image\|linux-headers'
```

Comment:
"sould" -> "should" but doesn't really matter lol

Reply:
lol thanks will update

Reply:
done